sympy/core/expr.py

```diff
@@ -1160,33 +1160,24 @@ def as_terms(self):
         gens, terms = set(), []
 
         for term in Add.make_args(self):
-            coeff, _term = term.as_coeff_Mul()
-
-            coeff = complex(coeff)
-            cpart, ncpart = {}, []
-
-            if _term is not S.One:
-                for factor in Mul.make_args(_term):
-                    if factor.is_number:
-                        try:
-                            coeff *= complex(factor)
-                        except (TypeError, ValueError):
-                            pass
-                        else:
-                            continue
 
-                    if factor.is_commutative:
-                        base, exp = decompose_power(factor)
+            _term, ncpart = term.args_cnc()
 
-                        cpart[base] = exp
-                        gens.add(base)
-                    else:
-                        ncpart.append(factor)
+            cpart = defaultdict(lambda: S.Zero)
 
-            coeff = coeff.real, coeff.imag
-            ncpart = tuple(ncpart)
+            # collect all Rational coefficients in coeff
+            coeff = S.One
+
+            for factor in _term:
+                if factor.is_Number:
+                    coeff *= factor
+                    continue
+                base, exp = decompose_power(factor)
+                cpart[base] += exp
```

Comment:
This change will have an impact on any unevaluated Mul in which there was a power with the same base raised to different powers.